pyiceberg/schema.py

```diff
@@ -1273,6 +1273,102 @@ def primitive(self, primitive: PrimitiveType) -> PrimitiveType:
         return primitive
 
 
+# Implementation copied from Apache Iceberg repo.
+def make_compatible_name(name: str) -> str:
+    if not _valid_avro_name(name):
+        return _sanitize_name(name)
+    return name
+
+
+def _valid_avro_name(name: str) -> bool:
+    length = len(name)
+    assert length > 0, "Empty name"
+    first = name[0]
+    if not (first.isalpha() or first == '_'):
+        return False
+
+    for i in range(1, length):
+        character = name[i]
+        if not (character.isalnum() or character == '_'):
+            return False
+    return True
+
+
+def _sanitize_name(name: str) -> str:
+    length = len(name)
+    sb = []
+    first = name[0]
+    if not (first.isalpha() or first == '_'):
+        sb.append(_sanitize_char(first))
+    else:
+        sb.append(first)
+
+    for i in range(1, length):
+        character = name[i]
+        if not (character.isalnum() or character == '_'):
+            sb.append(_sanitize_char(character))
+        else:
+            sb.append(character)
```

Comment:
We can also use the slice here:
```suggestion
    for character in name[1:]:
        if not (character.isalnum() or character == '_'):
            sb.append(_sanitize_char(character))
        else:
            sb.append(character)
```
Or:
```suggestion
    for character in name[1:]:
        sb.append(_sanitize_char(character) if not (character.isalnum() or character == '_') else character)
```
Or:
```suggestion
    sb = [first] + [
        _sanitize_char(character) if not (character.isalnum() or character == '_') else character
    ]
```